data-structures-and-algorithms/lesson01.rst

```diff
@@ -4,6 +4,15 @@ Data Structures and Algorithm Analysis - Lesson 1 Learning Objectives
 Goals
 -----
 
+- Understand how to express asymptotic worst-case runtime for an algorithm using Big-O notation
+- Sort ordered collections with Insertion Sort
 
 Objectives
 ----------
+
+- Explain the formal definition of big-O. In other words, what does it mean to say that *f(x)* is *O(g(x))*? 
```

Comment:
I agree that this is sticky. We need them to understand what expressions like O(n^2) mean, so the idea of a mathematical function is impossible to _completely_ avoid (even if we don't mention the formal definition or use symbols like f(x)). 

A potential rewording of this might be: "...what does it mean to say that the runtime of a function is O(_expression/function of n_), where n is the size of the input to the function."

Reply:
the way i worded it in the intro was

> ...in the form `O(complexity(n))`.